modules/core/src/lib/layer.js

```diff
@@ -476,15 +478,22 @@ export default class Layer extends Component {
     const cacheSize = pickingColorCache.length / 3;
 
     if (cacheSize < numInstances) {
+      let overAlloc;
+      if (numInstances > MAX_PICKING_COLOR_CACHE_SIZE) {
+        log.warn('Index out of picking color range. Picking might not work as expected.');
+        overAlloc = 1;
```

Comment:
This doesn't seem right. So if `numInstance` is `MAX_CACHE_SIZE - 1`, we still allocate twice as much? Shouldn't it cap the max array length instead?

Reply:
✓